web/core/src/jsMain/kotlin/org/jetbrains/compose/web/attributes/SyntheticEventListener.kt

```diff
@@ -0,0 +1,156 @@
+package org.jetbrains.compose.web.attributes
+
+import org.jetbrains.compose.web.events.SyntheticInputEvent
+import androidx.compose.web.events.SyntheticDragEvent
+import androidx.compose.web.events.SyntheticEvent
+import androidx.compose.web.events.SyntheticMouseEvent
+import androidx.compose.web.events.SyntheticWheelEvent
+import org.jetbrains.compose.web.attributes.EventsListenerBuilder.Companion.CHANGE
+import org.jetbrains.compose.web.attributes.EventsListenerBuilder.Companion.INPUT
+import org.jetbrains.compose.web.attributes.EventsListenerBuilder.Companion.SELECT
+import org.jetbrains.compose.web.events.*
+import org.w3c.dom.DragEvent
+import org.w3c.dom.TouchEvent
+import org.w3c.dom.clipboard.ClipboardEvent
+import org.w3c.dom.events.*
+
+open class SyntheticEventListener<T : SyntheticEvent<*>>(
+    val event: String,
+    val options: Options,
+    val listener: (T) -> Unit
+) : EventListener {
+
+    @Suppress("UNCHECKED_CAST")
+    override fun handleEvent(event: Event) {
+        listener(SyntheticEvent<EventTarget>(event) as T)
+    }
+}
+
+class Options {
+    // TODO: add options for addEventListener
+
+    companion object {
+        val DEFAULT = Options()
+    }
+}
+
+internal class AnimationEventListener(
+    event: String,
+    options: Options,
+    listener: (SyntheticAnimationEvent) -> Unit
+) : SyntheticEventListener<SyntheticAnimationEvent>(
+    event, options, listener
+) {
+    override fun handleEvent(event: Event) {
+        listener(SyntheticAnimationEvent(event))
+    }
+}
+
+internal class MouseEventListener(
+    event: String,
+    options: Options,
+    listener: (SyntheticMouseEvent) -> Unit
+) : SyntheticEventListener<SyntheticMouseEvent>(event, options, listener) {
+    override fun handleEvent(event: Event) {
+        listener(SyntheticMouseEvent(event.unsafeCast<MouseEvent>()))
```

Comment:
let's choose one of two - either we use unsafeCast or as

Reply:
`unsafeCast` is now used in this and similar cases